src/redux/slices/transactions/index.js

```diff
@@ -1,7 +1,103 @@
+import set from 'lodash-es/set';
 import { createSelector } from 'reselect';
 
+import { getTransactions, transactionExtraInfo } from '../../../utils/explorer-api';
+
 const SLICE_NAME = 'transactions';
 
+const initialState = {
+    transactions: {
+        byAccountId: {}
+    },
+    status: {
+        
+    }
+};
+
+const fetchTransactions = createAsyncThunk(
+    `${SLICE_NAME}/fetchTransactions`,
+    async ({ accountId }, { dispatch, getState }) => {
+        const transactions = await getTransactions({ accountId });
+
+        const { actions: { addTransactions, updateTransactions } } = transactionsSlice;
+
+        !selectTransactionsByAccountId(getState(), { accountId }).length
+            ? dispatch(addTransactions({ transactions, accountId }))
+            : dispatch(updateTransactions({ transactions, accountId }));
+    }
+);
+
+const fetchTransactionStatus = createAsyncThunk(
+    `${SLICE_NAME}/fetchTransactionStatus`,
+    async ({ hash, signer_id, accountId }, { dispatch, getState }) => {
+        let status;
+        try {
+            const transactionDetails = await transactionExtraInfo({ hash, signer_id });
+            status = Object.keys(transactionDetails.status)[0];
+        } catch (error) {
+            status = 'notAvailable';
+        }
+        const checkStatus = ['SuccessValue', 'Failure'].includes(status);
+
+        const { actions: { updateTransactionStatus } } = transactionsSlice;
+        dispatch(updateTransactionStatus({ status, checkStatus, accountId, hash }));
+    }
+);
+
+const transactionsSlice = createSlice({
+    name: SLICE_NAME,
+    initialState,
+    reducers: {
+        addTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+            set(state, ['transactions', 'byAccountId', accountId], transactions);
+        },
+        updateTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+
+            const transactionsState = state.transactions.byAccountId[accountId];
+            const hash = transactionsState.map((t) => t.hash_with_index);
+
+            transactions.forEach((t) => {
+                if (!hash.includes(t.hash_with_index)) {
+                    transactionsState.unshift(t);
+                    if (transactionsState.length > 10) {
+                        transactionsState.pop();
+                    }
+                }
+            });
+        },
+        updateTransactionStatus(state, { payload }) {
+            const { status, checkStatus, accountId, hash } = payload;
+
+            const transactionsState = state.transactions.byAccountId[accountId];
+            const index = transactionsState.findIndex((t) => t.hash === hash);
+            if (index !== -1) {
+                transactionsState[index].status = status;
+                transactionsState[index].checkStatus = checkStatus;
+            }
+        }
+    },
+    extraReducers: ((builder) => {
```

Comment:
We're missing a case for `rejected` -- right now, if the `fetchTransactions` thunk throws an error, the `loading` state will remain stuck set to `true` (because `fulfilled` only fires on _success_)

See example here: https://github.com/near/near-wallet/blob/9b79d5efb5f5f90e514c0b2eb33b2d915f3fbcfc/src/reducers/nft/index.js#L132-L133 for attaching the rejected handler

Reply:
Done.